test/new-e2e/tests/windows/service-test/startstop_test.go

```diff
@@ -0,0 +1,470 @@
+		),
+		awsHostWindows.WithAgentClientOptions(
+			agentclientparams.WithSkipWaitForAgentReady(),
+		),
+	))}
+	e2e.Run(t, s, opts...)
+}
+
+type baseStartStopSuite struct {
+	e2e.BaseSuite[environments.WindowsHost]
+	startAgentCommand func(host *components.RemoteHost) error
+	stopAgentCommand  func(host *components.RemoteHost) error
+}
+
+// TestAgentStartsAllServices tests that starting the agent starts all services (as enabled)
+func (s *baseStartStopSuite) TestAgentStartsAllServices() {
+	s.startAgent()
+	s.requireAllServicesState("Running")
+}
+
+// TestAgentStopsAllServices tests that stopping the agent stops all services
+func (s *baseStartStopSuite) TestAgentStopsAllServices() {
+	host := s.Env().RemoteHost
+
+	// run the test multiple times to ensure the agent can be started and stopped repeatedly
+	N := 10
+	if testing.Short() {
+		N = 1
+	}
+
+	for i := 1; i <= N; i++ {
+		s.T().Logf("Test iteration %d/%d", i, N)
+
+		s.startAgent()
+		s.requireAllServicesState("Running")
+
+		// stop the agent
+		err := s.stopAgentCommand(host)
+		s.Require().NoError(err, "should stop the datadogagent service")
+
+		// ensure all services are stopped
+		s.requireAllServicesState("Stopped")
+
+		// ensure there are no errors in the event log from the agent services
+		entries, err := s.getAgentEventLogErrorsAndWarnings()
+		s.Require().NoError(err, "should get agent errors and warnings from Application event log")
+		s.Require().Empty(entries, "should not have errors or warnings from agents in the event log")
+	}
+
+	// check event log for N sets of start and stop messages from each service
+	for _, serviceName := range s.expectedUserServices() {
+		providerName := serviceName
+		entries, err := windowsCommon.GetEventLogEntriesFromProvider(host, "Application", providerName)
+		s.Require().NoError(err, "should get event log entries from %s", providerName)
+		// message IDs from pkg/util/winutil/messagestrings
+		startingMessages := windowsCommon.Filter(entries, func(entry windowsCommon.EventLogEntry) bool {
+			return entry.ID == 7
+		})
+		startedMessages := windowsCommon.Filter(entries, func(entry windowsCommon.EventLogEntry) bool {
+			return entry.ID == 3
+		})
+		stoppingMessages := windowsCommon.Filter(entries, func(entry windowsCommon.EventLogEntry) bool {
+			return entry.ID == 12
+		})
+		stoppedMessages := windowsCommon.Filter(entries, func(entry windowsCommon.EventLogEntry) bool {
+			return entry.ID == 4
+		})
+		s.Assert().Len(startingMessages, N, "should have %d starting message in the event log", N)
+		s.Assert().Len(startedMessages, N, "should have %d started message in the event log", N)
+		s.Assert().Len(stoppingMessages, N, "should have %d stopping message in the event log", N)
+		s.Assert().Len(stoppedMessages, N, "should have %d stopped message in the event log", N)
+	}
+}
+
+func (s *baseStartStopSuite) SetupSuite() {
+	if setupSuite, ok := any(&s.BaseSuite).(suite.SetupAllSuite); ok {
+		setupSuite.SetupSuite()
+	}
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
This should do it:
```suggestion
	s.BaseSuite.SetupSuite()
```